pkg/build/options.go

```diff
@@ -45,6 +46,15 @@ func WithDisabledOptimizations() Option {
 	}
 }
 
+// WithGoReleaserBuildConfig is a functional option for providing build
+// settings based on a GoReleaser style configuration.
+func WithGoReleaserBuildConfig(buildCfgOverrides map[string]config.Build) Option {
```

Comment:
I think this is impossible for external dependents to use because `config.Build` is `internal/`. (cc @halvards)

We may want to have a `pkg/config` (or something) that exposes the usable subset of goreleaser's `config.Build`, then do some translation to talk advantage of their stuff.

Reply:
Good point. Thanks. I moved it to `pkg/config`. However, I am wondering whether maybe something like `pkg/ref/goreleaser/config` would be better, because `pkg/config` is pretty generic.

Reply:
@jonjohnsonjr Did you have to think about the name of package?

Reply:
Sorry, missed this comment. That's a good point.

My intuition is that this would only affect build config, so it might make more sense to flip the name and call this `build.Config` instead of `config.Build`?

I'm indifferent re: config vs configuration, but this seems reasonable enough:

```
builder.Build(..., build.WithConfig(build.Config{}))
```

We already have `build.Result` so it fits in well. If we want to have a separate configuration for e.g. the publisher stuff (currently set by KO_DOCKER_REPO), that could be mirrored as `publisher.Config`. I think scoping the configuration to the relevant package is probably best, but I'd be interested in other ideas.

Sorry for the back and forth on this :P

Reply:
No problem, not at all. I am here to learn. Naming things can be hard and `build.Config` makes perfect sense while also improves the readability of the code. I will change that right away.